mailpoet/lib/Subscription/ManageSubscriptionFormRenderer.php

```diff
@@ -134,6 +140,9 @@ private function getCustomFields(SubscriberEntity $subscriber): array {
 
       $customField['params']['value'] = $customFieldValues[$customFieldEntity->getId()] ?? null;
 
+      // Historically, custom fields were not sanitized so we need to sanitize them here
+      $customField = $this->dataSanitizer->sanitizeBlock($customField);
```

Comment:
It is not clear to me why we are sanitizing on output. Shouldn't we escape instead of sanitize?

As a rule of thumb, I think it is preferable to escape as late as possible. With that in mind, have you considered escaping the checkbox value inside the checkbox renderer (https://github.com/mailpoet/mailpoet/blob/f9356281b9aa0563e36e0c765c1f9c8c12f0a713/mailpoet/lib/Form/Block/Checkbox.php#L69
https://github.com/Automattic/mailpoet-security-vulnerabilities/blob/f9356281b9aa0563e36e0c765c1f9c8c12f0a713/mailpoet/lib/Form/Block/Checkbox.php#L69)?

Reply:
Thanks for the additional information, @costasovo. I had missed that we want to allow some HTML in the text displayed next to the checkbox. I like your solution. When you say calling wp_kses in the renderer, do you mean in \MailPoet\Form\Block\Checkbox::render()? Just checking as this is where I believe this code should go.

Reply:
Yes I want to use it in \MailPoet\Form\Block\Checkbox::render()! Thanks for the feedback. I'll update the PR.